staging/src/k8s.io/apimachinery/pkg/util/net/util.go

```diff
@@ -54,3 +54,20 @@ func IsConnectionReset(err error) bool {
 	}
 	return false
 }
+
+// Returns if the given err is "connection refused" error
+func IsConnectionRefused(err error) bool {
+	if urlErr, ok := err.(*url.Error); ok {
+		err = urlErr.Err
+	}
+	if opErr, ok := err.(*net.OpError); ok {
```

Comment:
this and the next if block can be chained using else, right ?

Reply:
no, you could get a net.OpError that is not wrapped by url.Error etc..